app/(auth)/logout/LogoutButton.tsx

```diff
@@ -0,0 +1,30 @@
+import { cookies } from 'next/headers';
+import { deleteSessionByToken } from '../../../database/sessions';
+import styles from './LogoutButton.module.scss';
+
+export default function LogoutButtonClient() {
+  return (
+    <form>
+      <button
+        className={styles.logoutButton}
+        /* @ts-ignore */
+        formAction={async () => {
+          'use server';
+          const cookieStore = cookies();
+          const token = cookieStore.get('sessionToken');
+
+          if (token) {
+            await deleteSessionByToken(token.value);
+          }
+
+          (cookies() as any).set('sessionToken', '', {
```

Comment:
- [ ] let's add an explanation (with official reference link) as a code comment too, so that we know what is happening here - why this "revalidates" / "refreshes the router" (not sure which one to use - we need to choose one official way of referring to this too)
- [ ] probably `as unknown as ResponseCookies` is the correct type
 
 
https://github.com/vercel/next.js/blob/062fbf6d2d1629ffa240b87af35955c51f5eaeaa/packages/next/src/server/async-storage/request-async-storage-wrapper.ts#LL42C4-L42C19

  until Shu fixes it:

  - https://github.com/vercel/next.js/issues/49259#issuecomment-1540019323